packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6688,31 +6688,36 @@ abstract class RenderObjectElement extends Element {
   }
 
   void _updateParentData(ParentDataWidget<ParentData> parentDataWidget) {
-    bool applyParentData = true;
-    assert(() {
-      try {
-        if (!parentDataWidget.debugIsValidRenderObject(renderObject)) {
-          applyParentData = false;
+    if (parentDataWidget.debugIsValidRenderObject(renderObject)) {
+      parentDataWidget.applyParentData(renderObject);
+    } else {
+      // If it's not a valid render object, then don't show the ErrorWidget, or
+      // throw an exception, just report the error and move on.
+      //
+      // This prevents it from crashing in release mode and still reports the
+      // original error in debug mode.
+      assert(() {
+        try {
           throw FlutterError.fromParts(<DiagnosticsNode>[
-            ErrorSummary('Incorrect use of ParentDataWidget.'),
+            ErrorSummary('Incorrect use of ParentDataWidget. The widget `${
+              parentDataWidget.runtimeType
```

Comment:
Calling `runtimeType.toString()` in release mode is expensive. Consider using the `objectRuntimeType()` helper, although that one also uses an assert.

See https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/foundation/object.dart

Reply:
I found a `toStringShort` helper function within this file which calls `objectRuntimeType`